### ISSUE TITLE
Allow column unioning in Tables.columns

### ISSUE THREAD

**Issue by bkamins** [JuliaData/Tables.jl#346]
Following the discussion in https://github.com/JuliaData/DataFrames.jl/pull/3372#discussion_r1335238316 It would make sense to add `cols` kwarg to `Tables.columns`.

Let us discuss here if it would make sense, because maybe it is better to have a column-oriented equivalent of `DictRowTable` as a separate object.

CC @nalimilan @quinnj

**Comment by bkamins** [comment]
@nalimilan - I have realized that we have `Tables.dictcolumntable` to accompany `Tables.dictrowtable` so this issue can be closed. I will update DataFrames.jl accordingly.

**Comment by nalimilan** [comment]
Good. It would still make sense to allow opting into a stricter behavior than the default (which I find dangerous) using `cols=:setequal` or `cols=:orderequal`. But this is lower priority.

**Comment by bkamins** [comment]
Yes - You mean something like `Tables.columns(table, cols=:orderequal)`? @quinnj - is it clear what we have in mind here?

**Comment by quinnj** [comment]
Closing because `Tables.dictcolumntable` already provides the requested column-unioning behavior. A stricter equality mode can be a separate focused proposal.
